### ISSUE TITLE
Linking via an alias `--alias` forces the use of `_URL` as the environment variable suffix

### ISSUE THREAD

**Issue by suwyn** [dokku/dokku-redis#234]
## Description of problem

When linking a postgres database to an app using the `--link-flags` argument, the plugin appends `_URL` to the environment variable name specified in the argument.

This causes an issue when the application you are linking does not use `_URL` as the suffix of the DSN URL.   In our case that is Metabase, which uses [`MB_DB_CONNECTION_URI`](https://www.metabase.com/docs/latest/configuring-metabase/environment-variables#mb_db_connection_uri)

### Steps to Reproduce

1. Create an application.
2. Create a postgres database.
3. Link the postgres database to the app by specifying an alias. `dokku postgres:link metabase metabase -a MB_DB_CONNECTION_URI`

#### Actual Results
Sets the environment variable `MB_DB_CONNECTION_URI_URL` on the metabase application

#### Expected Results

`MB_DB_CONNECTION_URI` would have been set.

The work around is to set the environment variable via the app config.  e.g. `dokku config:set metabase MB_DB_CONNECTION_URI=$(dokku postgres:info metabase --dsn)`

**Comment by josegonzalez** [comment]
I'll post the potential fix here (from a [discussion in discord](https://discord.com/channels/952684157123837952/1214125695819714622)):

TL;DR: We need a new flag called `--env-name` that tracks the alternative env-name. If set, we use that, otherwise, fallback to existing behavior.

- and adding a new flag to track the env-name value: https://github.com/dokku/dokku-redis/blob/master/common-functions#L701
- add a new `config_set` call to all the places where we currently do that in this file: https://github.com/dokku/dokku-redis/blob/master/common-functions#L635
    - you'd only want to add the `config_set` call if there is an `SERVICE_ENV_NAME` environment variable value
- running bin/generate (assuming you updated the docblock on the `link` subcommand [here](https://github.com/dokku/dokku-redis/blob/master/subcommands/link)) would be enough
- add a bats test here: https://github.com/dokku/dokku-redis/blob/master/tests/service_link.bats

That should make it generic enough that this is a copy-paste process for me to get this implemented on all datastores. You could implement the above on any datastore and I can take it from there, no need to focus on redis specifically.

---

This has been requested a few times and as I don't want to break existing behavior, I think the above is a good balance between desired functionality and backwards compatibility.
